### PR TITLE
Check for graveler *and* catalog ErrNotFound

### LINKED ISSUE #4427: lakeFS returns code 500 when getting absent object
Hi,

My locally-deployed lakeFS instance (`treeverse/lakefs:0.82.0`) returns a code 500 when trying to get an object that does not exist through the S3 API. This breaks compatibility with other applications.

Per boto3's documentation:
> You need the relevant read object (or version) permission for this operation. For more information, see [Specifying Permissions in a Policy](https://docs.aws.amazon.com/AmazonS3/latest/dev/using-with-s3-actions.html). If the object you request does not exist, the error Amazon S3 returns depends on whether you also have the ``s3:ListBucket`` permission.
> * If you have the ``s3:ListBucket`` permission on the bucket, Amazon S3 will return an HTTP status code 404 ("no such key") error. 
> * If you don’t have the ``s3:ListBucket`` permission, Amazon S3 will return an HTTP status code 403 ("access denied") error.

Expected behaviour: 
* lakeFS returns a code 404 as I have all read and write access.

### PR DESCRIPTION
**Pull request by arielshaqed**
Fixes #4427.

Opened #4435 to refactor to help protect against similar recurrences.

### CODE REVIEW

**Comment by arielshaqed** [comment]
Tested manually:
```
❯ AWS_MAX_ATTEMPTS=1 aws --profile local-s3 --endpoint http://localhost:8000/ s3api get-object --bucket foo --key main/asdf -

An error occurred (NoSuchKey) when calling the GetObject operation: The specified key does not exist.
```
(also tested with a bad branch and a bad repo).

Will add separately unit tests for getobject and headobject.

File: esti/s3_gateway_test.go [hunk 1]
```diff
@@ -114,6 +114,60 @@ func TestS3UploadAndDownload(t *testing.T) {
...
+		if string(got) != contents {
+			t.Errorf("Got contents \"%s\" but expected \"%s\"", string(got), contents)
+		}
+	})
+	t.Run("Doesn't exist", func(t *testing.T) {
+		_, err := client.GetObject(ctx, repo, goodPath, minio.GetObjectOptions{})
+		if err != nil {
+			t.Errorf("client.GetObject(%s, %s): %s", repo, goodPath, err)
+		}
+	})
```
Comment (Jonathan-Rosenberg) [inline esti/s3_gateway_test.go:168]: shouldn't this test check for the correct response if the path doesn't exist?

Comment (arielshaqed) [reply esti/s3_gateway_test.go]: It does, now :-)

**Review by Jonathan-Rosenberg — APPROVED** [review]
LGTM!

**Review by itaiad200 — APPROVED** [review]
Thanks for the quick fix!

**Comment by arielshaqed** [comment]
Thanks!

**Pull request merged**